src/view/observer/focusobserver.js

```diff
@@ -8,6 +8,10 @@
  */
 
 import DomEventObserver from './domeventobserver';
+import global from '@ckeditor/ckeditor5-utils/src/dom/global';
+
+const setTimeout = global.window.setTimeout;
+const clearTimeout = global.window.clearTimeout;
```

Comment:
In these particular tests I'd propose not to stub anything and have behaviour closest to the reality. The easier these tests will be the better because this is the most fragile part of our testing sute.